### PR TITLE
owner -> owner()

### PR DESCRIPTION
**Pull request by jeffwashington**
Problem
prepare for further abstraction of AccountSharedData

Summary of Changes
.owner -> .owner()

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -3254,7 +3254,7 @@ impl AccountsDb {
         let mut hasher = Hasher::default();
 
         hasher.hash(&account.data());
-        hasher.hash(&account.owner.as_ref());
+        hasher.hash(&account.owner().as_ref());
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:3257]: Doesn't `.owner()` already return a ref?

Comment (jeffwashington) [reply runtime/src/accounts_db.rs:3257]: as_ref on the pubkey causes it to return a &[u8], which is what hasher wants.
The leading & appears to be unnecessary but acceptable.

Comment (jeffwashington) [reply runtime/src/accounts_db.rs:3257]: in pubkey.rs:
impl AsRef<[u8]> for Pubkey {
    fn as_ref(&self) -> &[u8] {
        &self.0[..]
    }
}

Comment (jeffwashington) [reply runtime/src/accounts_db.rs:3257]: This was a great question, though.

Comment (brooksprumo) [reply runtime/src/accounts_db.rs:3257]: Gotcha, thanks!

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**